ocrd_network/ocrd_network/processing_server.py

```diff
@@ -670,3 +700,116 @@ async def list_processors(self) -> List[str]:
+            mets_path: str,
+            page_id: str,
+            agent_type: str = 'worker',
+    ) -> List[PYJobOutput]:
+        file_group_cache = {}
+        responses = []
+        for task in tasks:
+            # Find dependent jobs of the current task
+            dependent_jobs = []
+            for input_file_grp in task.input_file_grps:
+                if input_file_grp in file_group_cache:
+                    dependent_jobs.append(file_group_cache[input_file_grp])
+            # NOTE: The `task.mets_path` and `task.page_id` is not utilized in low level
+            # Thus, setting these two flags in the ocrd process workflow file has no effect
+            job_input_data = PYJobInput(
+                processor_name=task.executable,
+                path_to_mets=mets_path,
+                input_file_grps=task.input_file_grps,
+                output_file_grps=task.output_file_grps,
+                page_id=page_id,
+                parameters=task.parameters,
+                agent_type=agent_type,
+                depends_on=dependent_jobs,
+            )
+            response = await self.push_processor_job(
+                processor_name=job_input_data.processor_name,
+                data=job_input_data
+            )
+            for file_group in task.output_file_grps:
+                file_group_cache[file_group] = response.job_id
+            responses.append(response)
+        return responses
+
+    async def run_workflow(
+            self,
+            workflow: UploadFile,
+            mets_path: str,
+            agent_type: str = 'worker',
+            page_id: str = None,
+            page_wise: bool = False,
+            workflow_callback_url: str = None
+    ) -> PYWorkflowJobOutput:
+        # core cannot create workspaces by api, but processing-server needs the workspace in the
+        # database. Here the workspace is created if the path available and not existing in db:
+        # from pudb import set_trace; set_trace()
+        db_workspace = await db_create_workspace(mets_path)
+        if not db_workspace:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                                detail=f"Mets file not existing: {mets_path}")
+
+        workflow = (await workflow.read()).decode("utf-8")
+        try:
+            tasks_list = workflow.splitlines()
+            tasks = [ProcessorTask.parse(task_str) for task_str in tasks_list if task_str.strip()]
+        except BaseException as e:
+            print(e)
+            raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                                detail=f"Error parsing tasks: {e}")
+
+        if page_id:
+            page_range = expand_page_ids(page_id)
+        else:
+            # If no page_id is specified, all physical pages are assigned as page range
+            page_range = get_ocrd_workspace_physical_pages(mets_path=mets_path)
+        compact_page_range = f'{page_range[0]}..{page_range[-1]}'
```

Comment:
That seems unnecessary, you can just pass `None` as page_id to get the same behavior.

Reply:
Ah, no wait, you're right, this is for the database, which does not know how many pages there are in the workspace.

Reply:
Yes. With the `comapct_page_rage` the `page_id` is now always passed to lower levels with all pages when `page_id` is not set.